### PR TITLE
Update changelog with missing changes, closes #816

### LINKED ISSUE #816: Nothing in 0.14 changelog about layout scale changes
0.14 includes a bunch of changes to the layout scales: https://github.com/mapbox/assembly/commit/1946422f86f344d71ccd3b15da0c348595d49dd0#diff-759c953ebbf2b1a9d86f29ef197f9e30. These are going to cause a lot of trouble for us if we aren't super careful about them when we upgrade. @samanpwbb can you document those?

### PR DESCRIPTION
**Pull request by samanpwbb**
@davidtheclark for review

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## 0.14.0
 
+- Adds new 'limiter' class for setting max page width across screen sizes.
+- Significantly refactors layout scale to be more consistent across different properties and to increment more predictibly.
```
Comment (davidtheclark) [inline CHANGELOG.md]: I'd appreciate mention of the specific numbers that were added and removed. I got stumped by `py48` no longer working, and had to dig into the commit history to figure out what happened.

Comment (samanpwbb) [reply CHANGELOG.md]: I didn't want to do this because the documentation covers the specific changes. But if you think it'll be helpful, I will!

Comment (davidtheclark) [reply CHANGELOG.md]: I always find it really helpful for the changelog to point out the specific changes ("added classes x, y, z", "removed classes x, y, z"), so I can use the changelog entry as an "upgrade guide" — telling me exactly what I need to look for and alter in my code before upgrading.

**Comment by samanpwbb** [comment]
@davidtheclark back to you again.

**Review by davidtheclark — APPROVED** [review]
🍰

**Comment by davidtheclark** [comment]
Thanks!

**Pull request merged**